iaso/tasks/export_mobile_app_setup_for_user.py

```diff
@@ -0,0 +1,282 @@
+def _extract_filename_from_url(url):
+    return urlparse(url).path.split("/")[-1]
+
+
+def _call_endpoint(iaso_client, url, filename):
+    logger.info(f"{filename}: GET {url}")
+    result = iaso_client.get(url)
+
+    if isinstance(result, dict) and "count" in result:
+        logger.info(f"\tTotal count: {result['count']}")
+    if isinstance(result, dict) and "pages" in result:
+        logger.info(f"\tTotal pages: {result['pages']}")
+
+    return result
+
+
+def _download_form_attachments(iaso_client, tmp_dir, resources, app_id):
+    if len(resources) == 0:
+        return
+
+    # create subfolder for downloaded files
+    os.makedirs(os.path.join(tmp_dir, "formattachments"))
+    for resource in resources:
+        form_id = resource["form_id"]
+        os.makedirs(os.path.join(tmp_dir, "formattachments", str(form_id)))
+        url = resource["file"]
+        filename = _extract_filename_from_url(url)
+
+        logger.info(f"\tDOWNLOAD {url}")
+        attachment_file = None
+        # S3 urls contain a signature and don't work with additional auth headers
+        if "s3.amazonaws" in url:
+            attachment_file = requests.get(url)
+        else:
+            attachment_file = requests.get(url, headers=iaso_client.headers)
+
+        logger.info(f"\tDOWNLOAD manifest")
+        manifest_file = requests.get(
+            SERVER + f"/api/forms/{form_id}/manifest/?app_id={app_id}",
+            headers=iaso_client.headers,
+        )
+
+        with open(os.path.join(tmp_dir, "formattachments", str(form_id), filename), mode="wb") as f:
+            f.write(attachment_file.content)
+        # For the manifest.xml, rewrite the `downloadUrl` to the local file path
+        with open(os.path.join(tmp_dir, "formattachments", str(form_id), "manifest.xml"), mode="w") as f:
+            content = manifest_file.content.decode("utf-8")
+            url_regex = r"(?<=<downloadUrl>)(.*?)(?=</downloadUrl>)"
+            download_url = re.search(url_regex, content).group()
+            # don't use _extract_filename_from_url to preserve subpath
+            new_download_url = "formattachments/" + urlparse(download_url).path.split("/form_attachments/")[-1]
+            f.write(re.sub(url_regex, new_download_url, content))
+
+
+def _download_form_versions(iaso_client, tmp_dir, form_versions):
+    if len(form_versions) == 0:
+        return
+
+    os.makedirs(os.path.join(tmp_dir, "forms"))
+    for form_version in form_versions:
+        url = form_version["file"]
+        filename = _extract_filename_from_url(url)
+
+        logger.info(f"\tDOWNLOAD {url}")
+        response = None
+        # S3 urls contain a signature and don't work with additional auth headers
+        if "s3.amazonaws" in url:
+            response = requests.get(url)
+        else:
+            response = requests.get(url, headers=iaso_client.headers)
+
+        with open(os.path.join(tmp_dir, "forms", filename), mode="wb") as f:
+            f.write(response.content)
+
```

Comment:
This could be eventually put in a function and reused on lines 246 -> 255

Reply:
👍 done, I think I was a bit lazy because of the small difference with the url/path in case of a non S3 url